### PR TITLE
Fixes for compiling on NVHPC and NoOpComm

### PR DESCRIPTION
**Pull request by jacobmerson**

### CODE REVIEW

File: redev.cpp [hunk 1]
```diff
@@ -48,13 +49,13 @@ namespace redev {
     MPI_Gather(&len,1,MPI_INT,degree.data(),1,MPI_INT,root,comm);
     if(root==rank) {
       auto offset = redev::LOs(commSize+1);
-      std::exclusive_scan(degree.begin(), degree.end(), offset.begin(), redev::LO(0));
+      redev::exclusive_scan(degree.begin(), degree.end(), offset.begin(), redev::LO(0));
       auto allSerialized = redev::LOs(offset.back());
       MPI_Gatherv(serialized.data(), len, MPI_INT, allSerialized.data(),
-          degree.data(), offset.data(), MPI_INT, root, MPI_COMM_WORLD);
+          degree.data(), offset.data(), MPI_INT, root, comm);
```
Comment (cwsmith) [inline redev.cpp:55]: Whoops.  Sorry about this.

File: redev.h [hunk 2]
```diff
@@ -485,5 +500,9 @@ BidirectionalComm<T> Redev::CreateAdiosClient(std::string_view name, adios2::Par
   REDEV_ALWAYS_ASSERT(false);  //we should never get here
   return {nullptr, nullptr}; //silence compiler warning
 }
+template <typename T>
+BidirectionalComm<T> Redev::CreateNoOpClient() {
```
Comment (cwsmith) [inline redev.h:504]: I assume there would be no need for a 'NoOpServer'?

File: redev_exclusive_scan.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#ifndef REDEV_EXCLUSIVE_SCAN_H
+#define REDEV_EXCLUSIVE_SCAN_H
+namespace redev {
+  template <typename InputIt, typename OutputIt, typename T>
+  OutputIt exclusive_scan(InputIt first, InputIt last, OutputIt d_first, T init) {
```
Comment (cwsmith) [inline redev_exclusive_scan.h:5]: NVHPC/pgi compilers don't provide `std::exclusive_scan`?

**Review by cwsmith — APPROVED** [review]
LGTM.

File: redev_exclusive_scan.h (hunk 3 above)
Comment (jacobmerson) [reply redev_exclusive_scan.h:5]: I'm using `NVHPC 21.7` on summit which has the corresponding XGC build working: https://xgc.pppl.gov/html/cmake_build_instructions.html

It doesn't seem to have inclusive/exclusive scan.

**Pull request merged**